File: blackduck-artifactory-common/src/main/java/com/synopsys/integration/blackduck/artifactory/BlackDuckArtifactoryProperty.java
/*
 * blackduck-artifactory-common
 *
 * Copyright (c) 2022 Synopsys, Inc.
 *
 * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
 */
package com.synopsys.integration.blackduck.artifactory;

import java.util.Set;

import org.apache.commons.collections.set.UnmodifiableSet;

public enum BlackDuckArtifactoryProperty {
    @Deprecated
    BLACKDUCK_ORIGIN_ID("originId"),
    @Deprecated
    BLACKDUCK_FORGE("forge"),
    BLACKDUCK_PROJECT_NAME("projectName"),
    BLACKDUCK_PROJECT_VERSION_NAME("projectVersionName"),
    CRITICAL_VULNERABILITIES("criticalVulnerabilities"),
    HIGH_VULNERABILITIES("highVulnerabilities"),
    MEDIUM_VULNERABILITIES("mediumVulnerabilities"),
    LOW_VULNERABILITIES("lowVulnerabilities"),
    POLICY_STATUS("policyStatus"),
    POLICY_SEVERITY_TYPES("policySeverityTypes"),
    COMPONENT_VERSION_URL("componentVersionUrl"),
    @Deprecated
    COMPONENT_NAME_VERSION("componentNameVersion"),
    COMPONENT_VERSION_ID("componentVersionId"),
    PROJECT_VERSION_UI_URL("uiUrl"),
    OVERALL_POLICY_STATUS("overallPolicyStatus"),
    LAST_INSPECTION("lastInspection"),
    INSPECTION_STATUS("inspectionStatus"),
    INSPECTION_STATUS_MESSAGE("inspectionStatusMessage"),
    LAST_UPDATE("lastUpdate"),
    UPDATE_STATUS("updateStatus"),
    SCAN_TIME("scanTime"),
    SCAN_RESULT("scanResult"),
    SCAN_RESULT_MESSAGE("scanResultMessage"),
    POST_SCAN_ACTION_STATUS("postScanActionStatus"),
    POST_SCAN_PHASE("postScanPhase"),
    INSPECTION_RETRY_COUNT("inspectionRetryCount"),
    SCAAAS_SCAN_STATUS("scaaas.scanStatus"),
    SCAAAS_POLICY_STATUS("scaaas.policyStatus"),
    SCAAAS_FAILED_COUNT("scaaas.scanFailedCount"),
    SCAAAS_LAST_UPDATE("scaaas.lastUpdate"),
    SCAAAS_RESULTS_URL("scaaas.resultsUrl"),
    SCAAAS_SCAN_FAILURE_MESSAGE("scaaas.scanFailureMessage"),
    SCAAAS_VIOLATING_POLICY_RULES("scaaas.violatingPolicyRules");

    private final String propertyName;
    private final String timeName;

    private static final Set<BlackDuckArtifactoryProperty> scaaasProperties = Set.of(
            SCAAAS_SCAN_STATUS,
            SCAAAS_POLICY_STATUS,
            SCAAAS_FAILED_COUNT,
            SCAAAS_LAST_UPDATE,
            SCAAAS_RESULTS_URL,
            SCAAAS_SCAN_FAILURE_MESSAGE,
            SCAAAS_VIOLATING_POLICY_RULES);

    BlackDuckArtifactoryProperty(String suffix) {
        propertyName = "blackduck." + suffix;
        timeName = propertyName + ".converted";
    }

    public String getPropertyName() {
        return propertyName;
    }

    public String getTimeName() {
        return timeName;
    }

    public static Set<BlackDuckArtifactoryProperty> getScanAsAServiceProperties() {
        return scaaasProperties;
    }
}

File: blackduck-artifactory-common/src/main/kotlin/com/synopsys/integration/blackduck/artifactory/BlackDuckArtifactoryProperty.kt
/**
 * blackduck-artifactory-common
 *
 * Copyright (c) 2019 Synopsys, Inc.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package com.synopsys.integration.blackduck.artifactory

enum class BlackDuckArtifactoryProperty(suffix: String) {
    BLACKDUCK_ORIGIN_ID("originId"),
    BLACKDUCK_FORGE("forge"),
    BLACKDUCK_PROJECT_NAME("projectName"),
    BLACKDUCK_PROJECT_VERSION_NAME("projectVersionName"),
    HIGH_VULNERABILITIES("highVulnerabilities"),
    MEDIUM_VULNERABILITIES("mediumVulnerabilities"),
    LOW_VULNERABILITIES("lowVulnerabilities"),
    POLICY_STATUS("policyStatus"),
    POLICY_SEVERITY_TYPES("policySeverityTypes"),
    COMPONENT_VERSION_URL("componentVersionUrl"),
    PROJECT_VERSION_UI_URL("uiUrl"),
    OVERALL_POLICY_STATUS("overallPolicyStatus"),
    LAST_INSPECTION("lastInspection"),
    INSPECTION_STATUS("inspectionStatus"),
    INSPECTION_STATUS_MESSAGE("inspectionStatusMessage"),
    LAST_UPDATE("lastUpdate"),
    UPDATE_STATUS("updateStatus"),
    SCAN_TIME("scanTime"),
    SCAN_RESULT("scanResult"),
    SCAN_RESULT_MESSAGE("scanResultMessage"),
    POST_SCAN_ACTION_STATUS("postScanActionStatus"),
    POST_SCAN_PHASE("postScanPhase"),
    INSPECTION_RETRY_COUNT("inspectionRetryCount");

    val propertyName: String = "blackduck.$suffix"
    val timeName: String = "$propertyName.converted"
}
